test/NetTopologySuite.Samples.Console/Technique/SplitPolygonAtDateline.cs

```diff
@@ -0,0 +1,802 @@
+                    Debug.Print("resGeoms were fixed");
+
+                    // method 2
+                    //var pm = new PrecisionModel(10000000000.0);
+                    //resGeoms = new GeometryPrecisionReducer(pm).Reduce(resGeoms);
+                }
+            }
+
+            // print final vertices of all geometries
+            DebugPrintCoords(resGeoms.Coordinates.ToList(), "Final number of coordinates");
+
+            // success so return generated geometry (it could me multi-part)
+            return resGeoms;
+        }
+
+        /// <param name="coordsInp">A list of coordinates that define a polyline or polygon.</param>
+        /// <param name="proj4WktInp">The projection of the input coordinates as a PROJ4 WKT string.</param>
+        /// <returns>
+        /// A Geometry class that contains the result of converting the coordinates into a Polyline or Polygon. The result could be multiple
+        /// geometries.
+        /// </returns>
+        public static Geometry ToPolyExOp(List<Coordinate> coordsInp, string proj4WktInp)
+        {
+            return ToPolyExOp(coordsInp, proj4WktInp, OgcGeometryType.Polygon, double.NaN, double.NaN, false);
+        }
+
+        /// <summary>
+        /// Normalizes the input longitude.
+        /// </summary>
+        /// <returns>A <strong>Longitude</strong> containing the normalized value.</returns>
+        /// <remarks>This function is used to ensure that an angular measurement is within the
+        /// allowed bounds of 0° and 180°. If a value of 360° or 720° is passed, a value of 0°
+        /// is returned since traveling around the Earth 360° or 720° brings you to the same
+        /// place you started.</remarks>
+        /// <see cref="DotSpatial.Positioning"/>
+        private static double Normalize(double _decimalDegrees)
+        {
+            // Is the value not a number, infinity, or already normalized?
+            if (double.IsInfinity(_decimalDegrees) || double.IsNaN(_decimalDegrees))
+                return _decimalDegrees;
+            // If we're off the eastern edge (180E) wrap back around from the west
+            if (_decimalDegrees > 180)
+                return (-180 + (_decimalDegrees % 180));
+            // If we're off the western edge (180W) wrap back around from the east
+            return _decimalDegrees < -180 ? (180 + (_decimalDegrees % 180)) : _decimalDegrees;
+        }
+
+        /// <summary>
+        /// Determines if a polygon crosses the dateline. This is slightly different than checking a LineString as the Polygon is
+        /// closed. So a segment exists between the last vertex and the first vertex.
+        /// </summary>
+        /// <param name="coordsInp">The input list of coordinates.</param>
+        /// <returns>True if the coordinates as a polygon cross the dateline, False otherwise.</returns>
+        private static bool IsPolygonCrossesDatelineEx(List<Coordinate> coordsInp)
```

Comment:
Again, this should work on a `CoordinateSequence`

Reply:
I will scope that out.

Reply:
This is a function internal to the process. I am converting the CoordnateSequence to a List<Coordinate> anyway since I require a bunch of List functionality.

Reply:
I think you are using onlu `List.Exists` method, that can be re-implemented easily for coordinate sequences

Reply:
I use more List functions than that I believe. If the CoordinateSequence does not provide IEnumerable then what are the alternatives?